tests/test_parse_serialize_resource.py

```diff
@@ -0,0 +1,74 @@
+# Copyright Mozilla Foundation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from importlib.resources import files
+from unittest import TestCase
+
+from moz.l10n.resource import Format, parse_resource, serialize_resource
+from moz.l10n.resource.data import Resource
+
+
+def get_source(filename: str) -> bytes:
+    return files("tests.data").joinpath(filename).read_bytes()
+
+
+class TesteParseResource(TestCase):
+    def test_named_files(self):
+        data = (
+            "accounts.dtd",
+            "angular.xliff",
+            "bug121341.properties",
+            "defines.inc",
+            "demo.ftl",
+            "foo.po",
+            "hello.xliff",
+            "icu-docs.xliff",
+            "messages.json",
+            "strings.xml",
+            "test.properties",
+            "xcode.xliff",
+        )
+        for file in data:
+            res = parse_resource(file, get_source(file))
+            assert isinstance(res, Resource)
+            list(serialize_resource(res))
+            list(serialize_resource(res, trim_comments=True))
```

Comment:
What's the purpose of the last two lines without an `assert`?

Reply:
They exhaust the generator. This ensures that the complete resource can be serialized without error, and not just the first line. Is there a more pythonic way of doing that?

Reply:
I see. Something like https://stackoverflow.com/a/42327075 perhaps? A comment would also work.